### DISCUSSION TITLE
adam image 2.0.0 / PICO-8 / "Unsupported video mode"

### DISCUSSION THREAD

**Discussion by ctshh** [eduardofilo/RG350_adam_image#265] [General]
I just installed the lates adam image 2.0.0 on my RG 280V. Runs fine, except the PICO-8 emulator. Trying to start any cartdridge I get, in green letters, the screen

```
Unsupported video mode
> 160x205, XRGB8888
```
![IMG_1667](https://user-images.githubusercontent.com/6263/176546792-b810ca06-2703-4b73-85d1-b5a3ac6cd6c5.jpg)

and it hangs, I have to return by pressing select & start.

Any idea what that might be and if I can change anything to get my beloved PICO-8 running?
Thank you, Christian.

**Comment by a-dekker** [comment]
Retroarch PICO8 doesn't work out-of-the-box here either (rg350m). I use alternative PICO8 engines like tac08 and fake08_gcw0 for now.

**Reply by ctshh** [reply]
Would you happen to have any links for those or a manual on how to integrate?

**Reply by a-dekker** [reply]
You have to change your simplemenu config (/media/data/local/home/.simplemenu/section_groups/home computers.ini)
```
[PICO-8]
execs = /media/data/apps/retroarch_rg350_PICO8.opk,/media/data/apps/tac08.opk,/media/data/apps/fake08_gcw0.opk
romDirs = /media/sdcard/roms/PICO8/
aliasFile = /media/data/local/home/.simplemenu/alias_PICO-8.txt
```
The first entry in the execs line should be the default, so you might want to change the order.

**Reply by Captain-Barge** [reply]
To be thorough, I'll be a little more explicit about all of the steps:

  - Download `fake08_gcw0.opk` from [here](https://github.com/jtothebell/fake-08/releases)
  - Put the file here on your SD card: `/media/data/apps/fake08_gcw0.opk`
  - Change the PICO-8 entry your simplemenu config `/media/data/local/home/.simplemenu/section_groups/home computers.ini` to what is shown below (important: make sure that `fake08_gcw0.opk` is listed first on the `execs` line)
```
[PICO-8]
execs = /media/data/apps/fake08_gcw0.opk,/media/data/apps/retroarch_rg350_PICO8.opk,/media/data/apps/tac08.opk
romDirs = /media/sdcard/roms/PICO8/
aliasFile = /media/data/local/home/.simplemenu/alias_PICO-8.txt
```

**Reply by CIPop** [reply]
Thank you @Captain-Barge for the solution!

I can either switch to tac08 (using select) or use above solution to make Pico-8 png play.

**Comment by a-dekker** [comment]
@eduardofilo any idea if RA PICO-8 should work in your view?

**Reply by eduardofilo** [reply]
Most of the games I have for PICO-8 work with the RetroArch core installed in Adam 2.0. They are p8 though. I don't know if the png's will give more or less problems. I remember that there were some problems and that's why I kept the standalone emulator tac08.

**Reply by Shatur** [reply]
It looks like if `p8.png` extension doesn't work. But if I rename to just `.p8` - it works fine.

**Reply by a-dekker** [reply]
It seems to be as simple as that indeed.

**Reply by Shatur** [reply]
But this emulator can't run all games. [The newly added fake-08](https://github.com/eduardofilo/RG350_adam_image/commit/b23d2bffce9820597ac4b2a7b594670dca8bea08) is better, it has better compatibility.

**Reply by eduardofilo** [reply]
About the problem with PNG's in RA core, see the issue #322 reported by @Shatur.

**Comment by fuji8bit** [comment]
@eduardofilo ,  thanks for the excellent work on adam for 280V  I just installed v. 2.1

Like the OP, I am trying to run Pico8 using the RA opk, and I get the same issue (Unsupported Video Mode 160x205,  XRGB8888)
It loads the game (even p8.png), but freezes with the above display.

I saw there were fixes for the RG350's "Unsupported Video Mode" problems by using Video Scaling/Filters.  Is there a fix for this using RA video settings?  I've tried with no luck.  

Fake-08 and Taco-08 work, but not very well.  I have purchased the Pico-08 binaries.

**Reply by AIdisneyprincess3000** [reply]
@eduardofilo I'm in the same boat as @fuji8bit. For the one game I want to play on my RG280V, I have sound issues with Taco-8 and missing text/color issues with Fake-08. I have purchased Pico-08 and played around but was never able to start a game with the RA opk due to the issue reported by @ctshh. By the way, this is my first foray into retro gaming consoles and I'm grateful for all the work people like you put into these CFWs.

**Comment by fuji8bit** [comment]
I tried doing the same thing AmberElec does, which is launch the pico-08 executable from the "Tools" menu (would be the "Native Games" menu in Adam.

Long story short, I think I answered my own question --- The RG280V uses a MIPS processor, and the other handhelds that run Pico-08 natively are ARM (RG351V and RGB30).  This makes sense since the Raspberry Pi Image is a 64 bit ARM binary (dynamically linked to SDL).  

So until Lexaloffle releases more builds (don't hold your breath), or Pico-08 goes open source ,  Fake-08 is probably the best hope for the 280V, and there's not much to do about it.
